main/src/ca/uwaterloo/flix/language/phase/Lowering.scala

```diff
@@ -832,6 +832,39 @@ object Lowering {
       val itpe = Types.mkProvenanceOf(extVarType, loc)
       LoweredAst.Expr.ApplyDef(defn.sym, argExps, List.empty, itpe, tpe, eff, loc)
 
+    case TypedAst.Expr.FixpointQueryWithSelect(exps, queryExp, selects, _, _, pred, tpe, eff, loc) =>
+      val loweredExps = exps.map(visitExp)
+      val loweredQueryExp = visitExp(queryExp)
+
+      // Compute the arity of the predicate symbol.
+      // The type is either of the form `Array[(a, b, c)]` or `Array[a]`.
+      val (_, targs) = Type.eraseAliases(tpe) match {
+        case Type.Apply(tycon, innerType, _) => innerType.typeConstructor match {
+          case Some(TypeConstructor.Tuple(_)) => (tycon, innerType.typeArguments)
+          case Some(TypeConstructor.Unit) => (tycon, Nil)
+          case _ => (innerType, List(innerType))
+        }
+        case t => throw InternalCompilerException(s"Unexpected non-foldable type: '${t}'.", loc)
+      }
+
+      val arity = selects.length
+
+      // Compute the symbol of the function.
```

Comment:
which function? This comment seems hardly worht it. Maybe it can be combined with the next comment.